features/storage/nvstore/source/nvstore.cpp

```diff
@@ -559,6 +585,12 @@ int NVStore::garbage_collection(uint16_t key, uint16_t flags, uint8_t owner, uin
         new_area_offset = next_offset;
     }
 
+    // Reallocate _offset_by_key with new size
```

Comment:
This also shouldn't be here for the same reason. It should be moved to `set_max_keys` right after GC.